akeneo-design-system/src/components/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,51 @@
+import React, {Ref, ReactElement} from 'react';
+import styled from 'styled-components';
+import {AkeneoThemedProps, getColor} from '../../theme';
+import {IconProps} from '../../icons';
+
+const Container = styled.button<{color: string} & AkeneoThemedProps>`
+  background: none;
+  border: none;
+  padding: 0;
+  display: inline-flex;
+  color: ${({color}) => getColor(color, 100)};
+  cursor: ${({disabled}) => (disabled ? 'not-allowed' : 'pointer')};
+  opacity: ${({disabled}) => (disabled ? 0.6 : 1)};
+`;
+
+type IconButtonProps = {
+  /**
+   * The Icon to display.
+   */
+  icon: ReactElement<IconProps>;
+
+  /**
+   * The size of the Icon component.
+   */
+  size?: number;
+
+  /**
+   * The color of the Icon component.
+   */
+  color?: string;
+
+  /**
+   * Whether or not button is disabled.
+   */
+  disabled?: boolean;
```

Comment:
On click is not documented on storybook

```suggestion
  disabled?: boolean;
  
      /**
     * Function called when the user clicks on the button or hit enter when focused.
     */
    onClick?: (event: SyntheticEvent) => void;
```